generator/src/lib.rs

```diff
@@ -2395,7 +2440,39 @@ pub fn write_source_code<P: AsRef<Path>>(vk_xml: &Path, src_dir: P) {
         })
         .collect();
 
-    let spec = vk_parse::parse_file_as_vkxml(vk_xml).expect("Invalid xml file.");
+    let mut spec = vk_parse::parse_file_as_vkxml(vk_xml).expect("Invalid xml file.");
+
+    for elem in &mut spec.elements {
+        match elem {
+            vkxml::RegistryElement::Definitions(definitions) => {
+                for el in &mut definitions.elements {
+                    match el {
+                        vkxml::DefinitionsElement::Struct(_struct) => {
+                            for mem in &mut _struct.elements {
+                                match mem {
+                                    vkxml::StructElement::Member(field) => {
+                                        if matches!(field.array, Some(vkxml::ArrayType::Dynamic))
+                                            && field
+                                                .size
+                                                .as_ref()
+                                                // TODO: Should rather check if no identifier is a member variable
+                                                .map_or(false, |s| s.contains('*'))
+                                        {
+                                            field.array = Some(vkxml::ArrayType::Static);
+                                        }
+                                    }
+                                    _ => {}
+                                }
+                            }
+                        }
+                        _ => {}
+                    }
+                }
+            }
+            _ => {}
+        }
+    }
```

Comment:
This entire bit is here because vk.xml does not contain something like `type_t (*myvariable)[sz]`, where `[` is the only way to get it detected as a static-length array. Instead, pointers to static sized arrays are represented as a _single_ pointer with static length (`len=` and `altlen=` fields in `vk.xml`).

It should be moved to vkxml/vk_parse.